### DISCUSSION TITLE
What's the dev workflow for making changes to nx-dotnet and seeing them in your nx workspace?

### DISCUSSION THREAD

**Discussion by photomoose** [nx-dotnet/nx-dotnet#89] [Q&A] [answered]
Hi @AgentEnder,

I've been trying to make some tweaks to nx-dotnet locally, but can't seem to get those changes integrated into my existing nx workspace. I managed to do it successfully once in the past, now I can't see to get it working.

Here's what I'm currently doing:

1. Forked and checked out nx-dotnet repo
2. Make changes to local repo
3. `yarn build` in nx-dotnet root
4. Change into `nx-dotnet/dist/packages/[core|dotnet|util]` folders and execute`yarn link`
5. In the root of my nx-workspace, execute `yarn link @nx-dotnet/[core|dotnet|util]`
6. Run standard nx commands against my workspace

I seem to be ending up with problems such as the following:

```
> nx run common-identity:build
Cannot find module '/Users/tom.davis/Source/nx-dotnet/packages/core/src/executors/build/executor'
Require stack:
- /Users/tom.davis/Contingent/services-dotnet/node_modules/@nrwl/tao/src/shared/workspace.js
- /Users/tom.davis/Contingent/services-dotnet/node_modules/@nrwl/tao/src/commands/run.js
- /Users/tom.davis/Contingent/services-dotnet/node_modules/@nrwl/tao/index.js
- /Users/tom.davis/Contingent/services-dotnet/node_modules/@nrwl/cli/lib/run-cli.js
```

Any suggestions would be most welcome. Thanks!

**Comment by AgentEnder** [comment, marked as answer]
The way I typically do dev for the project follows this flow:

1. Run `yarn publish-local` in the `nx-dotnet` repo.
2. Create a new nx workspace, with `create-nx-workspace` wherever you want to test your changes. If you are using an existing workspace for this, just make sure its git status is clean instead.
3. Copy `.npmrc.local` from the `nx-dotnet` repo to your project, rename it to `.npmrc`
4. Run `npm i --save-dev @nx-dotnet/core@99.99.99`
	- If it cannot find the version, the `.npmrc` file is probably not being found. 
5. Test changes

**Reply by Tungsten78** [reply]
‘publish-local’  has been replaced by ‘yarn sandbox’. 

See [contributing](https://github.com/nx-dotnet/nx-dotnet/blob/master/CONTRIBUTING.md#validating-plug-in-changes-locally)
